### PR TITLE
setPermissions/setData bug (fixes #91)

### LINKED ISSUE #91: bucket.setPermissions() deletes existing data!
The request has both attribute `data` and `permissions`, but shouldn't.

And `data` contains `id` and `last_modified` only. The server will overwrite the existing data and remove any custom attribute defined on the bucket :) 

This might also be true for `setData()` which shouldn't overwrite existing permissions.

### PR DESCRIPTION
**Pull request by leplatrem**
* [x] Reproduce bug with integration tests (ref #91)
* [x] Wait for #83 to be merged
* [x] Fix code in requests.js
* [x] Wait for https://github.com/Kinto/kinto/pull/662 to be merged

### CODE REVIEW

**Comment by leplatrem** [comment]
This is ready to review @n1k0 r?

File: src/requests.js [hunk 1]
```diff
@@ -20,28 +21,78 @@ function safeHeader(safe, last_modified) {
...
- */
-export function createBucket(bucketName, options={}) {
-  if (!bucketName) {
-    throw new Error("A bucket name is required.");
-  }
-  // Note that we simply ignore any "bucket" option passed here, as the one
-  // we're interested in is the one provided as a required argument.
-  const { data={}, headers, permissions, safe } = {
+
+function _create(path, {data, permissions}, options={}) {
```
Comment (n1k0) [inline src/requests.js]: Nit: as we're already in a module and these functions are not exported, it's probably not necessary to prefix them with an underscore.

Comment (leplatrem) [inline src/requests.js]: yeah but `delete` is a reserved keyword :/

Comment (n1k0) [inline src/requests.js]: Oh, good point :o let's keep them that way then.

File: src/requests.js [hunk 2]
```diff
@@ -20,28 +21,78 @@ function safeHeader(safe, last_modified) {
...
+    safe,
+    patch,
+  } = {...requestDefaults, ...options};
+  const { last_modified } = { ...data, ...options };
+
+  if (Object.keys(omit(data, "id", "last_modified")).length === 0) {
+    data = undefined;
+  }
+
+  return removeUndefined({
```
Comment (n1k0) [inline src/requests.js]: I'm surprised this is needed as `undefined` properties are removed by serialization while it's already performed when sending the request body through `execute`: https://github.com/Kinto/kinto-client/blob/master/src/base.js#L366-L369

Am I missing something here? What breaks when this call is removed?

**Comment by n1k0** [comment]
Looks good, just a remaining question I'd like to have an answer :)

**Comment by leplatrem** [comment]
I'll merge this, because I took the comment into account in #99

**Pull request merged**